src/pgen/cluster/agn_feedback.cpp

```diff
@@ -349,7 +362,32 @@ void AGNFeedback::FeedbackSrcTerm(parthenon::MeshData<parthenon::Real> *md,
                 "Kinetic injection leads to temperature below jet and existing gas");
 #endif
           }
+
+          // Apply velocity ceiling
+          const Real v2 =
+              SQR(prim(IV1, k, j, i)) + SQR(prim(IV2, k, j, i)) + SQR(prim(IV3, k, j, i));
```

Comment:
Should we pull out the `eos.ConsToPrim(cons, prim, nhydro, nscalars, k, j, i);` from the `#ifndef NDEBUG` above so that it's always called now (given that we here now rely on the updated `prim` values?

Reply:
I think that's a good idea. We'll be checking the criteria for the production runs then.

On another note, I need to check whether the application  `cons` and `prim` is correct throughout these chained source terms. If we're always updating to `cons` but anywhere use that updated data to compute the strength of another source term, I think we'd be creating an inconsistency where the order we apply these source terms would change the result